llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -117,9 +117,13 @@ bool llvm::isVectorIntrinsicWithScalarOpAtArg(Intrinsic::ID ID,
                                               unsigned ScalarOpdIdx) {
   switch (ID) {
   case Intrinsic::abs:
+  case Intrinsic::vp_abs:
   case Intrinsic::ctlz:
+  case Intrinsic::vp_ctlz:
   case Intrinsic::cttz:
+  case Intrinsic::vp_cttz:
   case Intrinsic::is_fpclass:
+  case Intrinsic::vp_is_fpclass:
```

Comment:
Do you still need these changes?

Reply:
Comments for this?

Reply:
The second parameter of this intrinsic is a scalar, which should remain in scalar form after vectorization.
`llvm.vp.abs.nxv1i8(<vscale x 1 x i8> %va, i1 false, <vscale x 1 x i1> splat (i1 true), i32 %evl)`

Reply:
And where exactly it is used, why do you need this change?

Reply:
![Uploading image.png…]()

Reply:
if not change the code,the second parament of vp.abs will splat: `<vscale x 4 x i1> shufflevector (<vscale x 4 x i1> insertelement (<vscale x 4 x i1> poison, i1 true, i64 0), <vscale x 4 x i1> poison, <vscale x 4 x i32> zeroinitializer)`